tests/test_keyval.py

```diff
@@ -1,11 +1,75 @@
+import os
 import pytest
-
+import sys
+import time
+from multiprocessing import Process
```

Comment:
Should we explicitly tell Processes to use `spawn` to start the processes instead of `fork` which according to [this](https://docs.python.org/3/library/multiprocessing.html#process-and-exceptions) might sometimes be the default.

Reply:
I think `fork` is more prone to making bugs appear, so it's probably better that it's the default on Linux.

Also, in the these tests we only open file descriptors after the `fork` (assuming the previous ones have been correctly closed), so it shouldn't impact these tests.

Reply:
`flock` locks are inherited when a process is forked, which might cause some tests to differ, but I suppose its only in the tests so it doesn't matter as much.

Reply:
With the current code we shouldn't be holding any locks when the `fork` calls are made.

Reply:
good point.